### ISSUE TITLE
Malformed html case related to html comment

### ISSUE THREAD

**Issue by slorber** [gregjacobs/Autolinker.js#88]
Hi,

Just want to mention this weird corner case I encoutered. I work with ReactJS which for HTML fragments requires absolutely well-formed content or it's completely messed up. 

I would be nice to know if we can safely trust your library to return well-formed HTML, as an error here could be fatal (in our case an user reported being unable to use anything for 1 week).
One solution is to automatically reparse output from the lib with the browser or jquery but it's not very performant.

Here's the case.
As you can see an HTML comment produced some weird output here:

```
<!<a href="http://--.post-author--" target="_blank">--.post-author--</a>>
```

Input:
```html
<img src="https://s.greenqloud.com/com.stample.s3/stample-1424113830303-49429dfa-867b-4ac1-b87a-37b741f7025c" alt="Les clés du succès d’Eventbrite – Une interview vidéo de son co-fondateur Renaud Visage"><div>
						   
<div>
	<div>
							<!-- BEGIN BREADCRUMBS-->
						<!-- END BREADCRUMBS -->
		<div>
		<h3> Video </h3> 
	</div>
	 
	 <span class="single_post_date"> 23 septembre 2013 </span> 	</div>
</div>

 
	 
	
				
				<!-- Post Content -->
		<div>	 
			<p><iframe src="http://www.youtube.com/embed/weYOd-D1mh4?feature=oembed"></iframe></p>
<p><a href="http://www.eventbrite.com/">Eventbrite</a>, c'est aujourd'hui le leader mondial de l'organisation d'événements en ligne, avec à date :</p>
<p>- 2 milliards de dollars de ventes de billets sur la plateforme<strong>dont 25% ces six derniers mois</strong>,</p>
<p>- 1,5 millions d'événements organisés,</p>
<p>- cette année un chiffre d'affaires d'1 milliard de dollars (ventes de billets),</p>
<p>- 100 000 organisateurs dans le monde.</p>
<p>&nbsp;</p>
<p><em><strong>Un Français dans la Silicon Valley</strong></em></p>
<p>Renaud Visage, le co-fondateur et CTO d'Eventbrite, est Français et partage son temps entre la Califormie, Londres, Paris et d'autres villes en Europe. Il nous a raconté lors d'une interview vidéo comment il a été amené à co-fonder Eventbrite et ses étapes de développement :&nbsp; les débuts, la période bootstrap, les levées de fond (un total de 140 millions de dollars), l'internationalisation.</p>
<p>Il nous a aussi exposé les nouveaux projets : toujours plus d'international, la découverte sociale, la verticalisation de l'offre et l'arrivée sur les réservations de spectacles, théâtre ou cinéma.</p>
<p>En fin d'interview Renaud nous livre quelques clés du succès d'Eventbrite. Un insight à retenir pour d'autres entreprises, et pas que les start-ups.</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<div></div>   
			<div></div>
		</div>
		<!-- //Post Content -->
		
		
	
	
	 
	
<!-- <div class="post-author clearfix">
	<h3 class="post-author_h"> </h3>
	<p class="post-author_gravatar"></p>
	<div class="post-author_desc">
	 
		<div class="post-author_link">
			<p>: </p>
		</div>
	</div>
</div><!--.post-author-->


												<div> 
							
							<ul> 
														 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2015/01/28/vessel-5-choses-a-savoir-sur-la-start-up-video-de-jason-killar-lex-ceo-de-hulu/"> Vessel : 5 choses à savoir sur la start-up vidéo de Jason Kilar, l’ex CEO de Hulu </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">26</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2015/01/28/vessel-5-choses-a-savoir-sur-la-start-up-video-de-jason-killar-lex-ceo-de-hulu/">
											
										</a>
									</div> 
									<p> 
										

Vessel &nbsp;créé par Jason Kilar (Ex-Hulu et Yahoo) a été lancée le 17 décembre auprès des créateurs de vidéos .

Comme le fait remarquer ... 
									</p>
									
									<div> </div>
								</li>
							 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2015/01/20/amazon-va-produire-des-films-de-cinema-et-raccourcir-les-delais-de-diffusion/"> Amazon va produire des films de cinéma et raccourcir les délais de diffusion </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">39</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2015/01/20/amazon-va-produire-des-films-de-cinema-et-raccourcir-les-delais-de-diffusion/">
											
										</a>
									</div> 
									<p> 
										

Amazon, déja producteur de la série TV "Transparent", récompensée aux Golden Globes vient d’annoncer son entrée dans la production cinémat... 
									</p>
									
									<div> </div>
								</li>
							 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2014/12/04/emily-bell-journalisme-et-silicon-valley-aux-sources-du-schisme/"> Emily Bell&nbsp;: Journalisme et Silicon Valley, aux sources du schisme </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">66</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2014/12/04/emily-bell-journalisme-et-silicon-valley-aux-sources-du-schisme/">
											
										</a>
									</div> 
									<p> 
										Emily Bell, ancienne Directrice du Guardian, dirige aujourd’hui&nbsp; le TOW Center for Digital Journalism à Columbia.

Elle livrait la semaine derni... 
									</p>
									
									<div> </div>
								</li>
														</ul>
							 
						</div> 

					</div> 
```

Output:
```html
<img src="https://s.greenqloud.com/com.stample.s3/stample-1424113830303-49429dfa-867b-4ac1-b87a-37b741f7025c" alt="Les clés du succès d’Eventbrite – Une interview vidéo de son co-fondateur Renaud Visage"><div>
						   
<div>
	<div>
							<!-- BEGIN BREADCRUMBS-->
						<!-- END BREADCRUMBS -->
		<div>
		<h3> Video </h3> 
	</div>
	 
	 <span class="single_post_date"> 23 septembre 2013 </span> 	</div>
</div>

 
	 
	
				
				<!-- Post Content -->
		<div>	 
			<p><iframe src="http://www.youtube.com/embed/weYOd-D1mh4?feature=oembed"></iframe></p>
<p><a href="http://www.eventbrite.com/">Eventbrite</a>, c'est aujourd'hui le leader mondial de l'organisation d'événements en ligne, avec à date :</p>
<p>- 2 milliards de dollars de ventes de billets sur la plateforme<strong>dont 25% ces six derniers mois</strong>,</p>
<p>- 1,5 millions d'événements organisés,</p>
<p>- cette année un chiffre d'affaires d'1 milliard de dollars (ventes de billets),</p>
<p>- 100 000 organisateurs dans le monde.</p>
<p>&nbsp;</p>
<p><em><strong>Un Français dans la Silicon Valley</strong></em></p>
<p>Renaud Visage, le co-fondateur et CTO d'Eventbrite, est Français et partage son temps entre la Califormie, Londres, Paris et d'autres villes en Europe. Il nous a raconté lors d'une interview vidéo comment il a été amené à co-fonder Eventbrite et ses étapes de développement :&nbsp; les débuts, la période bootstrap, les levées de fond (un total de 140 millions de dollars), l'internationalisation.</p>
<p>Il nous a aussi exposé les nouveaux projets : toujours plus d'international, la découverte sociale, la verticalisation de l'offre et l'arrivée sur les réservations de spectacles, théâtre ou cinéma.</p>
<p>En fin d'interview Renaud nous livre quelques clés du succès d'Eventbrite. Un insight à retenir pour d'autres entreprises, et pas que les start-ups.</p>
<p>&nbsp;</p>
<p>&nbsp;</p>
<div></div>   
			<div></div>
		</div>
		<!-- //Post Content -->
		
		
	
	
	 
	
<!-- <div class="post-author clearfix">
	<h3 class="post-author_h"> </h3>
	<p class="post-author_gravatar"></p>
	<div class="post-author_desc">
	 
		<div class="post-author_link">
			<p>: </p>
		</div>
	</div>
</div><!<a href="http://--.post-author--" target="_blank">--.post-author--</a>>


												<div> 
							
							<ul> 
														 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2015/01/28/vessel-5-choses-a-savoir-sur-la-start-up-video-de-jason-killar-lex-ceo-de-hulu/"> Vessel : 5 choses à savoir sur la start-up vidéo de Jason Kilar, l’ex CEO de Hulu </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">26</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2015/01/28/vessel-5-choses-a-savoir-sur-la-start-up-video-de-jason-killar-lex-ceo-de-hulu/">
											
										</a>
									</div> 
									<p> 
										

Vessel &nbsp;créé par Jason Kilar (Ex-Hulu et Yahoo) a été lancée le 17 décembre auprès des créateurs de vidéos .

Comme le fait remarquer ... 
									</p>
									
									<div> </div>
								</li>
							 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2015/01/20/amazon-va-produire-des-films-de-cinema-et-raccourcir-les-delais-de-diffusion/"> Amazon va produire des films de cinéma et raccourcir les délais de diffusion </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">39</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2015/01/20/amazon-va-produire-des-films-de-cinema-et-raccourcir-les-delais-de-diffusion/">
											
										</a>
									</div> 
									<p> 
										

Amazon, déja producteur de la série TV "Transparent", récompensée aux Golden Globes vient d’annoncer son entrée dans la production cinémat... 
									</p>
									
									<div> </div>
								</li>
							 
								<li> 
									
									<div>
										<a href="http://www.viuz.com/2014/12/04/emily-bell-journalisme-et-silicon-valley-aux-sources-du-schisme/"> Emily Bell&nbsp;: Journalisme et Silicon Valley, aux sources du schisme </a> 
										
			<span class="pw pw-widget ra1-pw-classicWidget ra1-pw_size_small pw-layout-horizontal pw-counter-horizontal">
				<a href="http://www.viuz.com/2013/09/23/les-cle-de-succes-deventbrite-une-interview-video-de-son-co-fondateur-renaud-visage/"><span class="pw-icon ra1-pw-icon ra1-pw-icon-post"></span><span class="pw-button-counter"><i></i><span class="pw-button-counter__count ra1-pw-button-counter-value">66</span></span></a>
			</span>  									</div>
									<div> 
																				<a href="http://www.viuz.com/2014/12/04/emily-bell-journalisme-et-silicon-valley-aux-sources-du-schisme/">
											
										</a>
									</div> 
									<p> 
										Emily Bell, ancienne Directrice du Guardian, dirige aujourd’hui&nbsp; le TOW Center for Digital Journalism à Columbia.

Elle livrait la semaine derni... 
									</p>
									
									<div> </div>
								</li>
														</ul>
							 
						</div> 

					</div> 
```

**Comment by gregjacobs** [comment]
Wow, great catch. Def not handling HTML comment tags correctly. Will see what I can do.

**Comment by gregjacobs** [comment]
Sorry for the super late reply btw.. =/

**Comment by gregjacobs** [comment]
Fixed in [v0.17.1](https://github.com/gregjacobs/Autolinker.js/releases/tag/0.17.1)
